tests/file.cpp

```diff
@@ -14,10 +14,41 @@ import tmp;
 #include <ostream>
 #include <utility>
 
+#ifdef WIN32
+#include <windows.h>
+#else
+#include <fcntl.h>
+#endif
+
 namespace tmp {
 
 namespace fs = std::filesystem;
 
+namespace {
+
+/// Checks if the given file handle is valid
+/// @param handle handle to check
+/// @returns @c true if the handle is valid, @c false otherwise
+bool native_handle_is_valid(file::native_handle_type handle) {
+#ifdef WIN32
+    BY_HANDLE_FILE_INFORMATION info;
+    return GetFileInformationByHandle(handle, &info);
+#else
+    return fcntl(handle, F_GETFD) != -1;
+#endif
+}
+}    // namespace
+
+/// Tests requirements for native_handle_type
+TEST(file, native_handle_type) {
+    // `TriviallyCopyable` named requirement
+    static_assert(std::is_trivially_copyable_v<file::native_handle_type>);
+#ifdef WIN32
+    // Confirm that `HANDLE` is `void*` as implemented in `file`
+    static_assert(std::is_same_v<HANDLE, void*>);
```

Comment:
Might be a good idea to move this to the `tmp.cpp` so that the project does not compile on the unsupported platforms even if the tests are not built